danger_code_ws/2.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <cstdlib>
+
+// Function to dynamically allocate a matrix of size rows x cols
+double** allocateMatrix(int rows, int cols) {
```

Comment:
Consider using `std::vector<std::vector<double>>` instead of manual dynamic memory allocation.  This will automatically handle memory allocation and deallocation, reducing the risk of memory leaks and simplifying the code.  This addresses potential errors related to forgetting to `deallocateMatrix` and potential issues with memory allocation failures.